diff --git a/modules/gatsby_endpoints/gatsby_endpoints.module b/modules/gatsby_endpoints/gatsby_endpoints.module
index cbf4b05..ee0b0c7 100644
--- a/modules/gatsby_endpoints/gatsby_endpoints.module
+++ b/modules/gatsby_endpoints/gatsby_endpoints.module
@@ -112,7 +112,7 @@ function gatsby_endpoints_form_alter(&$form, FormStateInterface $form_state, $fo
 
     // Override node edit form.
     if (!empty($preview_settings[$node_type]) && $preview_url) {
-      $alias = \Drupal::service('path.alias_manager')->getAliasByPath('/node/' . $node->id());
+      $alias = \Drupal::service('path_alias.manager')->getAliasByPath('/node/' . $node->id());
 
       // If this is the front-page we don't want to pass the alias
       // (as Gatsby will likely 404).
@@ -120,6 +120,9 @@ function gatsby_endpoints_form_alter(&$form, FormStateInterface $form_state, $fo
         $alias = '';
       }
 
+      $sites = array("/amp", "/bnp", "/cinp", "/corp", "/knp", "/ninp", "/pknp", "/uktnp");
+      $alias = str_replace($sites, "", $alias);
+
       // Add Gatsby Preview button.
       $form['actions']['gatsby_preview'] = [
         '#type' => 'button',
@@ -183,7 +186,10 @@ function gatsby_endpoints_node_view(array &$build, $entity, $display, $view_mode
         return;
       }
 
-      $alias = \Drupal::service('path.alias_manager')->getAliasByPath('/node/' . $node->id());
+      $alias = \Drupal::service('path_alias.manager')->getAliasByPath('/node/' . $node->id());
+
+      $sites = array("/amp", "/bnp", "/cinp", "/corp", "/knp", "/ninp", "/pknp", "/uktnp");
+      $alias = str_replace($sites, "", $alias);
 
       // If this is the front-page we don't want to pass the alias
       // (as Gatsby will likely 404).
diff --git a/modules/gatsby_endpoints/src/GatsbyEndpointTrigger.php b/modules/gatsby_endpoints/src/GatsbyEndpointTrigger.php
index aebfafe..f1818d3 100644
--- a/modules/gatsby_endpoints/src/GatsbyEndpointTrigger.php
+++ b/modules/gatsby_endpoints/src/GatsbyEndpointTrigger.php
@@ -111,7 +111,7 @@ class GatsbyEndpointTrigger {
       }
     }
 
-    $preview_path = "/__refresh";
+    $preview_path = "/";
     $preview_urls = $endpoint->getPreviewUrls();
     if (!empty($preview_urls) && !empty($preview_urls['preview_url'])) {
       foreach ($preview_urls['preview_url'] as $preview_url) {
@@ -163,7 +163,7 @@ class GatsbyEndpointTrigger {
       $json['secret'] = $secret;
     }
 
-    $preview_path = "/__refresh";
+    $preview_path = "/";
     $preview_urls = $endpoint->getPreviewUrls();
     if (!empty($preview_urls) && !empty($preview_urls['preview_url'])) {
       foreach ($preview_urls['preview_url'] as $preview_url) {
diff --git a/modules/gatsby_instantpreview/src/GatsbyInstantPreview.php b/modules/gatsby_instantpreview/src/GatsbyInstantPreview.php
index 4baed37..43a575b 100644
--- a/modules/gatsby_instantpreview/src/GatsbyInstantPreview.php
+++ b/modules/gatsby_instantpreview/src/GatsbyInstantPreview.php
@@ -76,7 +76,7 @@ class GatsbyInstantPreview extends GatsbyPreview {
       $json['secret'] = $settings->get('secret_key');
     }
 
-    $preview_path = "/__refresh";
+    $preview_path = "/";
     if ($settings->get('legacy_preview_url')) {
       // The legacy URL expects an encoded JSON string.
       $encoded_json = json_encode($this->bundleData('preview', '_legacy', $json));
@@ -137,7 +137,7 @@ class GatsbyInstantPreview extends GatsbyPreview {
       $json['secret'] = $settings->get('secret_key');
     }
 
-    $preview_path = "/__refresh";
+    $preview_path = "/";
     if ($settings->get('legacy_preview_url')) {
       // The legacy URL expects an encoded JSON string.
       $json = json_encode($json);
diff --git a/src/GatsbyPreview.php b/src/GatsbyPreview.php
index dc6f44a..e7ef46e 100644
--- a/src/GatsbyPreview.php
+++ b/src/GatsbyPreview.php
@@ -86,7 +86,7 @@ class GatsbyPreview {
     $preview_url = $settings->get('server_url');
 
     if ($preview_url) {
-      $this->updateData('preview', $preview_url, FALSE, "/__refresh");
+      $this->updateData('preview', $preview_url, FALSE, "/");
     }
 
     $incrementalbuild_url = $settings->get('incrementalbuild_url');
@@ -113,7 +113,7 @@ class GatsbyPreview {
       'action' => 'delete',
     ];
 
-    $preview_path = "/__refresh";
+    $preview_path = "/";
     $settings = $this->configFactory->get('gatsby.settings');
     $preview_url = $settings->get('server_url');
     if ($preview_url) {
